opentelekomcloud/data_source_opentelekomcloud_rds_flavors_v1.go

```diff
@@ -0,0 +1,115 @@
+package opentelekomcloud
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/huaweicloud/golangsdk/openstack/rds/v1/datastores"
+	"github.com/huaweicloud/golangsdk/openstack/rds/v1/flavors"
+)
+
+func dataSourceRdsFlavorV1() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourcedataSourceRdsFlavorV1Read,
+
+		Schema: map[string]*schema.Schema{
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"datastore_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"datastore_version": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"ram": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"speccode": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
This isn't optional as its not used to filter the data source, only computed.

Reply:
hi, this is used to filter the data source. So it is optional.